dimod/constrained.py

```diff
@@ -618,6 +619,88 @@ def from_file(cls, fp: Union[BinaryIO, ByteString]) -> "ConstrainedQuadraticMode
 
         return cqm
 
+    def iter_violations(self, sample_like, *, skip_satisfied: bool = False, clip: bool = True,
+                        ) -> Iterator[Tuple[Hashable, Bias]]:
+        """Yield constraint labels and the amount the constraints are violated.
+
+        Args:
+            sample_like: A sample over the CQM variables.
+            skip_satisfied: If True, samples that are not violated are skipped.
+            clip: If False, then negative violations will be returned.
+
+        Yields:
+            A 2-tuple containing the constraint label and the amount of
+            constraints violation.
```

Comment:
See above. One problem with this phrasing is it is confusing whether what is returned is the number of violations or the amplitude of each violation.